qiskit_nature/operators/second_quantization/vibrational_op.py

```diff
@@ -0,0 +1,338 @@
+
+            ops: List[VibrationalOp] = []
+            for dense_label, coeff in dense_labels:
+                new_op = reduce(lambda a, b: a @ b, (
+                        VibrationalOp((label, 1), num_modes, num_modals) for label in dense_label
+                ))
+                # We ignore the type here because mypy only sees the complex coefficient
+                ops.append(coeff * new_op)  # type: ignore
+
+            op = reduce(lambda a, b: a + b, ops)
+
+            self._labels = op._labels.copy()
+
+    def __repr__(self) -> str:
+        if len(self) == 1:
+            if self._coeffs[0] == 1:
+                return f"VibrationalOp('{self._labels[0]}')"
+            return f"VibrationalOp({self.to_list()[0]})"
+        return f"VibrationalOp({self.to_list()})"  # TODO truncate
+
+    def __str__(self) -> str:
+        """Sets the representation of `self` in the console."""
+        if len(self) == 1:
+            label, coeff = self.to_list()[0]
+            return f"{label} * {coeff}"
+        return "  " + "\n+ ".join([f"{label} * {coeff}" for label, coeff in self.to_list()])
+
+    def __len__(self):
+        return len(self._labels)
+
+    @property
+    def num_modes(self) -> int:
+        """The number of modes.
+        Returns:
+            The number of modes
+        """
+        return self._num_modes
+
+    @property
+    def num_modals(self) -> List[int]:
+        """The number of modals.
+        Returns:
+            The number of modals
+        """
+        return self._num_modals
+
+    @property
+    def register_length(self) -> int:
+        """Getter for the length of the register that the VibrationalOp `self` acts on."""
+        return self._register_length
+
+    def mul(self, other: complex) -> "VibrationalOp":
+        if not isinstance(other, (int, float, complex)):
+            raise TypeError(
+                f"Unsupported operand type(s) for *: 'VibrationalOp' and '{type(other).__name__}'"
+            )
+        return VibrationalOp(list(zip(self._labels, (other * self._coeffs).tolist())),
+                             self._num_modes, self._num_modals)
+
+    def add(self, other: "VibrationalOp") -> "VibrationalOp":
+        if not isinstance(other, VibrationalOp):
+            raise TypeError(
+                f"Unsupported operand type(s) for +: 'VibrationalOp' and '{type(other).__name__}'"
+            )
+
+        # Check compatibility
+        if self.num_modes != other.num_modes or self.num_modals != other.num_modals:
+            raise TypeError("Incompatible register lengths for '+'.")
+
+        return VibrationalOp(
+            list(
+                zip(self._labels + other._labels, np.hstack((self._coeffs, other._coeffs)).tolist())
+            ),
+            self._num_modes,
+            self._num_modals,
+        )
+
+    def to_list(self) -> List[Tuple[str, complex]]:
+        """Getter for the operator_list of `self`"""
```

Comment:
This is not a getter like the properties above. More a method to return the operator as a List of (label, coeff) tuples